src/services/codecovEventMetrics/useStoreCodecovEventMetric.tsx

```diff
@@ -0,0 +1,103 @@
+import { useMutation } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+
+import Api from 'shared/api'
+
+const mutationQuery = `
+  mutation storeEventMetric(
+    $input: StoreEventMetricsInput!
+  ) {
+    storeEventMetric(
+      input: $input
+    ) {
+      error {
+        ... on UnauthenticatedError {
+          message
+        }
+        ... on ValidationError {
+          message
+        }
+      }
+    }
+  }
+`
+
+interface Params {
+  provider: string
+}
+
+interface StoreEventMetricMutationArgs {
+  owner: string
+  event: string
+  jsonPayload: object
+}
+
+const LOCAL_STORAGE_KEY = 'UserOnboardingMetricsStored'
+const MAX_ENTRIES = 30
+
+const generateMetricString = (
+  owner: string,
+  event: string,
+  jsonPayload: string
+) => {
+  return `${owner}|${event}|${jsonPayload}`
+}
+
+const getStoredMetrics = (): string[] => {
+  const storedMetrics = localStorage.getItem(LOCAL_STORAGE_KEY)
+  return storedMetrics ? JSON.parse(storedMetrics) : []
+}
+
+const isMetricInLocalStorage = (metricString: string) => {
+  const storedMetrics = getStoredMetrics()
+  return storedMetrics.includes(metricString)
+}
+
+const addMetricToLocalStorage = (metricString: string) => {
+  let storedMetrics = getStoredMetrics()
+
+  // limit size of metrics stored to ensure we are in reasonable localstorage limits
+  if (storedMetrics.length >= MAX_ENTRIES) {
+    storedMetrics = storedMetrics.slice(1)
+  }
+
+  storedMetrics.push(metricString)
+  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(storedMetrics))
+}
+
+export const useStoreCodecovEventMetric = () => {
+  const { provider } = useParams<Params>()
+
+  const mutation = useMutation({
+    mutationFn: async ({
+      owner,
```

Comment:
Where did we end up on tracking owner and user? Last I can recall, we were going to track both, but not require either.

Reply:
We're just using org id at the moment(required). User can be supplied if need but is optional.